### ISSUE TITLE
Reboot

### ISSUE THREAD

**Issue by dbgnu76** [moroen/IKEA-Tradfri-plugin#118]
Is there any chance maybe for you to implement a reboot command to the gateway !
Like https://antoinealb.net/2019/08/10/automatically-rebooting-tradfri-gateway.html

**Comment by moroen** [comment]
That I can do, how would you like this in the domoticz interface? A push-button to reset the gateway? Such a button could be scheduled in the usual domoticz manner.

**Comment by basrieter** [comment]
That would be very much appreciated, especially as the gateway will become unresponsive after a longer period. Just rebooting it every week will work around that.

**Comment by Bolten88** [comment]
Cool. This would be nice!

**Comment by moroen** [comment]
The latest development version (0.9.15) adds a push button that reboots the gateway, this can be scheduled just like any other button.

I've made another change that I've been planning for quite a while, but been a bit reluctant to do, but it is about time.

Tradfricoap (the API for controlling the gateway) has  been separated out into it's own module, available on [pypi](https://pypi.org/project/tradfricoap/). On well configured systems where the interactive interpreter is the same as the one used by domoticz, the upgrade should be simple:

```
$ cd plugins/IKEA-Tradfri
$ rm -rf tradfricoap
$ sudo -H pip3 install tradfricoap
```

On systems where domoticz is unable to use system modules, tradfricoap can still be installed into the plugin directory with git:

```
$ cd plugins/IKEA-Tradfri
$ git clone https://github.com/moroen/tradfricoap.git
```

For systems using coapcmd, there is a new version of this as well:
```
$ cd plugins/IEKA-Tradfri
$ sh ./install_coapcmd.sh
```
or download the latest prebuild binaries.

**Comment by dbgnu76** [comment]
This is so great, and it is working like a charm :) Well Done :)

**Comment by basrieter** [comment]
Works like a charm here.

**Comment by Bolten88** [comment]
I've installed the latest version and upgraded tradfricoap. Everything works great. I think (but not sure) that it looks like it's reacts faster on on/off command. But it's hard to meassure when it is only a few miliseconds.

**Comment by moroen** [comment]
I've been running this version for a while my selv, and it seems to work. Merged into master, and closing issue.

**Comment by Bolten88** [comment]
@moroen thanks again for all the effort. I'm still running the development version because of these reasons:

1. battery monitoring issue #117 
2. Reboot button #118 
3. The upgrade to the new tradfricoap (#118 as well?)

If i'm understanding you correctly i could move to master again right?

**Comment by moroen** [comment]
@Bolten88 

Yes, you should be able to use the laster master, fixes those three issues have been merged into master, version 0.10.0
